### PR TITLE
[istio] Alt linux and Distroless powered images

### PR DESCRIPTION
**Pull request by skurbatov**
## Description
Migrate istio containers version 1.16 and 1.19 to Alt Linux and Distroless distribution.
Migrate kiali containers version 1.67 to Alt Linux distribution.

## Why do we need it, and what problem does it solve?
Using the Alt linux distribution in builds allows you to achieve a higher level of security.

## What is the expected result?

All images of the Istio module are based on either AltLinux or Distroless.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: istio
type: chore
summary: Migrate istio and kiali containers to Alt Linux and Distroless distribution
impact: All pods in the `d8-istio` namespace will be automatically restarted. Other pods with istio-sidecars will apply changes after a manual restart.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @skurbatov. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @skurbatov. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @skurbatov. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by Beastlex — APPROVED** [review]

**Review by AndreyPavlovFlant — APPROVED** [review]

File: modules/110-istio/images/kiali/Dockerfile [hunk 1]
```diff
@@ -1,14 +1,19 @@
+ARG BASE_ALT
 ARG BASE_ALPINE
 
 # original kiali
 FROM quay.io/kiali/kiali:v1.67.2@sha256:92fec2b44580519b922d37747114865df4cc4cd51a2fe8342f090c8941e6905d as artifact
 
-FROM $BASE_ALPINE
+FROM $BASE_ALPINE as artifact2
```
Comment (apolovov) [inline modules/110-istio/images/kiali/Dockerfile:7]: Don't need extra artifact.

```
adduser --no-create-home --uid 1000 --gid 1000 kiali2
...
```

File: modules/110-istio/images/api-proxy/Dockerfile [hunk 2]
```diff
@@ -1,11 +1,15 @@
-ARG BASE_ALPINE
+ARG BASE_ALT
 ARG BASE_GOLANG_16_ALPINE
 FROM $BASE_GOLANG_16_ALPINE as artifact
 WORKDIR /src/
 COPY main.go go.mod go.sum /src/
 RUN GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -ldflags="-s -w" -o api-proxy main.go
 
-FROM $BASE_ALPINE
+FROM $BASE_ALT
```
Comment (apolovov) [inline modules/110-istio/images/api-proxy/Dockerfile:8]: distroless pls.

**Review by apolovov — CHANGES_REQUESTED** [review]

File: modules/110-istio/images/api-proxy/Dockerfile (hunk 2 above)
Comment (skurbatov) [reply modules/110-istio/images/api-proxy/Dockerfile]: @apolovov the api-proxy, cni, operator migrated to Distroless.

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @skurbatov. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @skurbatov. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by apolovov — APPROVED** [review]

**Review by name212 — APPROVED** [review]

**Pull request merged**